src/nfa/operations.cc

```diff
@@ -56,6 +56,126 @@ namespace {
         return lts_for_simulation.compute_simulation();
     }
 
+    bool vec_contains(std::vector<std::pair<State, State>> vec, std::pair<State, State> par){
+        for (size_t i = 0; i < vec.size(); i++){
+            if (vec[i] == par){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    Nfa reduce_testing(const Nfa& aut){
+        // ! Preprocessing
```

Comment:
The pseudo code for this algorithm is "split" into section with these descriptions. I tried make clear which parts of the real code represent which parts of the pseudo code.

Reply:
Sounds good. Great idea. I would however use a more readable notation. Something like `// Pseudocode: Preprocessing.` or similar. Otherwise, everyone will be unsure of what the `// !` should represent.